src/utils/pgwire/src/net.rs

```diff
@@ -79,6 +80,13 @@ impl Listener {
             Self::Tcp(listener) => {
                 let (stream, addr) = listener.accept().await?;
                 stream.set_nodelay(true)?;
+                // Set TCP keepalive to 5 minutes, which is less than the connection idle timeout of 350 seconds in AWS ELB.
+                // https://docs.aws.amazon.com/elasticloadbalancing/latest/network/network-load-balancers.html#connection-idle-timeout
+                {
+                    let r = socket2::SockRef::from(&stream);
+                    let ka = socket2::TcpKeepalive::new().with_time(Duration::from_secs(300));
+                    r.set_tcp_keepalive(&ka)?;
+                }
```

Comment:
Interesting. Is this a hack or a common usage of `socket2`?

Reply:
I think it's the common usage and the only way for setting these extra parameters until https://github.com/tokio-rs/tokio/issues/3082 is addressed.